packages/editor/src/browser/editor.ts

```diff
@@ -237,6 +277,22 @@ export interface TextEditor extends Disposable, TextEditorSelection, Navigatable
     setEncoding(encoding: string, mode: EncodingMode): void;
 
     readonly onEncodingChanged: Event<string>;
+
+    /**
+     * Find all matches in an editor for the given options.
+     * @param options the options for finding matches.
+     *
+     * @returns the list of matches.
+     */
+    findMatches?(options: FindMatchesOptions): FindMatch[];
+
+    /**
+     * Get the text for a certain line.
+     * @param lineNumber the line number.
+     *
+     * @returns the text at the given line.
+     */
+    getLineContent?(lineNumber: number): string;
```

Comment:
I think we can do without this by using [`TextEditor.document.getLineContent(line)`](https://github.com/eclipse-theia/theia/blob/9e46acb5f9a0da7988832df395e2d5d166bd8cd6/packages/editor/src/browser/editor.ts#L31-L34)